miden-lib/asm/faucets/basic.masm

```diff
@@ -0,0 +1,68 @@
+use.miden::sat::account
+use.miden::sat::faucet
+use.miden::sat::tx
+
+#! Distributes freshly minted fungible assets to the provided address.
+#! 
+#! Inputs: [ASSET, tag, RECIPIENT, ...]
+#! Outputs: [note_ptr, 0, 0, 0, 0, 0, 0, 0, 0, ...]
+#!
+#! - ASSET is the fungible asset to be received.
+#! - tag is the tag to be included in the note.
+#! - RECIPIENT is the recipient of the note, i.e., 
+#!   hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash).
+#! - note_ptr is the pointer to the memory address in the kernel.
+#!   This cannot directly be accessed from another context.
+#!
+#! FAILS if: 
+#! - The asset being minted is not associated with the faucet the transaction is being executed
+#!   against.
+#! - The asset is not well formed.
+#! - The total issuance after minting is greater than the maximum amount 
+#!   allowed.
+export.distribute
+    # get max supply of this faucet. We assume max_supply is stored at pos 3 in memory slot 0
+    push.0 exec.account::get_item drop drop drop 
+    # => [max_supply, ASSET, tag, RECIPIENT, ...]
```

Comment:
I don't think we can use slot 0 for this info as if we want to use these methods with Falcon signature authentications, the slots will conflict. We should probably pick slot 1.

Also, I'd probably create a named constant to use with the `push` instruction.